src/components/course/data/utils.jsx

```diff
@@ -149,74 +149,102 @@ export function findCouponCodeForCourse(couponCodes, catalogList = []) {
+      return -1;
+    }
+    if (firstOfferValue >= secondOfferValue) {
+      return 1;
+    }
+  }
+  if (!firstOfferValue && secondOfferValue) {
+    return -1;
+  }
+  if (firstOfferValue && !secondOfferValue) {
+    return 1;
+  }
+  return 0;
+};
+
 /**
  * Returns an applicable enterprise offer to the specified enterprise catalogs, if one exists, with the
  * following prioritization:
  *   - Offer with no bookings limit (global or user)
  *   - Offer with user bookings limit
  *   - Offer with global bookings limit
+ *   - Offer with user enrollment limit
+ *   - Offer with global enrollment limit
  *
  * @param {array} enterpriseOffers List of enterprise offers available for the enterprise customer.
- * @param {array} catalogList List of enterprise catalog UUIDs associated with a given course.
  * @param {number} coursePrice The price of the course.
  *
  * @returns An object containing the metadata for the enterprise offer, if any, most applicable for
- * the specified enterporise catalog uuids and course price.
+ * the specified enterprise catalog uuids and course price.
  */
 export const findEnterpriseOfferForCourse = ({
-  enterpriseOffers, catalogList = [], coursePrice,
+  enterpriseOffers,
+  catalogsWithCourse,
+  coursePrice,
 }) => {
   if (!coursePrice) {
     return undefined;
   }
 
-  const applicableEnterpriseOffers = enterpriseOffers.filter((enterpriseOffer) => {
-    const {
-      remainingBalance,
-      remainingBalanceForUser,
-    } = enterpriseOffer;
-    const isCourseInCatalog = catalogList.includes(enterpriseOffer.enterpriseCatalogUuid);
-    if (!isCourseInCatalog) {
-      return false;
-    }
-    if (offerHasBookingsLimit(enterpriseOffer)) {
-      if (remainingBalance !== null && remainingBalance < coursePrice) {
+  const orderedEnterpriseOffers = enterpriseOffers
+    .filter((enterpriseOffer) => {
+      const isCourseInCatalog = catalogsWithCourse.includes(enterpriseOffer.enterpriseCatalogUuid);
+      if (!isCourseInCatalog) {
         return false;
       }
-
-      if (remainingBalanceForUser !== null && remainingBalanceForUser < coursePrice) {
-        return false;
+      return true;
+    })
+    .sort((a, b) => {
+      const isFirstOfferRedeemable = a.disabledEnrollReasonType === null;
+      const isSecondOfferRedeemable = b.disabledEnrollReasonType === null;
+      const firstOffer = a.enterpriseOffer;
+      const secondOffer = b.enterpriseOffer;
+
+      let comparison = 0;
+
+      if (isFirstOfferRedeemable && isSecondOfferRedeemable) {
+        comparison = compareOffersByProperty({ firstOffer, secondOffer, property: 'remainingApplications' });
+        comparison = compareOffersByProperty({ firstOffer, secondOffer, property: 'remainingApplicationsForUser' });
+        comparison = compareOffersByProperty({ firstOffer, secondOffer, property: 'remainingBalance' });
+        comparison = compareOffersByProperty({ firstOffer, secondOffer, property: 'remainingBalanceForUser' });
```

Comment:
What is the behavior we're shooting for here?  As it is, only remainingBalanceForUser will be considered for the final comparison.

Reply:
See the docstring of this function:

```
/**
 * Returns an applicable enterprise offer to the specified enterprise catalogs, if one exists, with the
 * following prioritization:
 *   - Offer with no bookings limit (global or user)
 *   - Offer with no applications limit (global or user)
 *   - Offer with user bookings limit
 *   - Offer with global bookings limit
 *   - Offer with user enrollment limit
 *   - Offer with global enrollment limit
 */
```

I agree there's something about this implementation that feels off.